src/main/java/com/iexec/common/docker/DockerRunRequest.java

```diff
@@ -72,30 +73,37 @@ public String getStringArgsCmd() {
         return devices != null ? new ArrayList<>(devices) : List.of();
     }
 
-    // override builder's isSgx() & devices() methods
+    public SgxDriverMode getSgxDriverMode() {
+        return sgxDriverMode != null ? sgxDriverMode : SgxDriverMode.NONE;
+    }
+
+    // override builder's sgxDriverMode() & devices() methods
     public static class DockerRunRequestBuilder { 
-        private boolean isSgx;
+        private SgxDriverMode sgxDriverMode;
         private List<Device> devices;
 
         /**
          * Add an SGX device when isSgx is true.
          * 
-         * @param isSgx
+         * @param sgxDriverMode SGX driver mode
          * @return
          */
-        public DockerRunRequestBuilder isSgx(boolean isSgx) {
-            this.isSgx = isSgx;
-            if (!isSgx) {
+        public DockerRunRequestBuilder sgxDriverMode(SgxDriverMode sgxDriverMode) {
+            this.sgxDriverMode = sgxDriverMode;
+            if (!SgxDriverMode.isDriverModeNotNone(sgxDriverMode)) {
                 return this;
             }
             if (this.devices == null) {
                 this.devices = new ArrayList<>();
             }
-            Device sgxDevice = new Device(
-                    SgxUtils.SGX_CGROUP_PERMISSIONS,
-                    SgxUtils.SGX_DEVICE_PATH,
-                    SgxUtils.SGX_DEVICE_PATH);
-            this.devices.add(sgxDevice);
+
+            for (String deviceName : sgxDriverMode.getDevices()) {
+                Device sgxDevice = new Device(
+                        SgxUtils.SGX_CGROUP_PERMISSIONS,
+                        "/dev/" + deviceName,
+                        "/dev/" + deviceName);
+                this.devices.add(sgxDevice);
```

Comment:
```suggestion
                this.devices.add(Device.parse("/dev/" + deviceName));
```
`Device.parse` has the same result has `new Device("rwm", "device", "device")`

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-common/pull/308/commits/24b0c04479f5ce6a5963a0c09882a81f478e7c41.